recipes/opencv/4.x/conanfile.py

```diff
@@ -0,0 +1,393 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.29.1"
+
+
+class OpenCVConan(ConanFile):
+    name = "opencv"
+    license = "Apache-2.0"
+    homepage = "https://opencv.org"
+    description = "OpenCV (Open Source Computer Vision Library)"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("computer-vision", "deep-learning", "image-processing")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "contrib": [True, False],
+               "parallel": [False, "tbb", "openmp"],
+               "with_jpeg": [False, "libjpeg", "libjpeg-turbo"],
+               "with_png": [True, False],
+               "with_tiff": [True, False],
+               "with_jpeg2000": [False, "jasper", "openjpeg"],
+               "with_openexr": [True, False],
+               "with_eigen": [True, False],
+               "with_webp": [True, False],
+               "with_gtk": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "parallel": False,
+                       "contrib": False,
+                       "with_jpeg": "libjpeg",
+                       "with_png": True,
+                       "with_tiff": True,
+                       "with_jpeg2000": "jasper",
+                       "with_openexr": True,
+                       "with_eigen": True,
+                       "with_webp": True,
+                       "with_gtk": True}
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _contrib_folder(self):
+        return "contrib"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.gtk
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and \
+           "MT" in str(self.settings.compiler.runtime) and self.options.shared:
+            raise ConanInvalidConfiguration("Visual Studio and Runtime MT is not supported for shared library.")
```

Comment:
```suggestion
            raise ConanInvalidConfiguration("Visual Studio and Runtime MT is not supported for shared library.")
        if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) < "4":
            raise ConanInvalidConfiguration("Clang 3.x can build OpenCV 3.x due an internal bug.")
```

Same error: https://github.com/conan-io/conan-center-index/pull/3465#issuecomment-725459253